esp_jpeg/tjpgd/tjpgd.c

```diff
@@ -906,7 +975,10 @@ static JRESULT mcu_output (
                         if (ix == 8) {
                             py += 64 - 8;    /* Jump to next block if double block heigt */
                         }
-                        pc += ix & 1;               /* Step forward chroma pointer every two pixels */
+                        /* Step forward chroma pointer every two pixels */
+                        if (ix % 2) {
+                            pc++;
+                        }
```

Comment:
I'm offering a beer to anyone how can explain why test 2 is not passing without this change!

Reply:
I might be missing something, but on first look this appears to be a compiler bug:

```
# a5 is pc, a4 is ix
=> 0x42007aca <+266>:	extui	a5, a4, 0, 1
   0x42007acd <+269>:	addx2	a5, a5, a5
```

`extui` overwrites the value in a5, so `pc` is zero after this instruction. The correct code would look like this:

```
  extui	rd, a4, 0, 1
  addx2	a5, a5, rd
```

where `rd` is another register (not a4 or a5).

Reply:
I think we can resolve this for now. I am running c-reduce to make a smaller reproducer for the issue before reporting this to toolchain team.